### DISCUSSION TITLE
Set the Dialog with dynamic height and width

### DISCUSSION THREAD

**Discussion by bearyung** [AvaloniaUtils/DialogHost.Avalonia#53] [Q&A] [answered]
Currently I can define height and width of the Dialog's content, is there any way to set that have a dynamic height and width, e.g. it always has 20 margin, so when the dialog is shown and when user resize the windows, the content of the dialog can be resize with it.

I've tried to set the margin of the content, the DialogMargin in <dialogHostAvalonia:DialogHost/> and the padding  but it has no effect.

Thanks for the help.

**Comment by SKProCH** [comment, marked as answer]
Hello, @bearyung 

Seems like this is can be dome via positioners. This is a custom classes which can arrange dialoghost popup in some ways. E.g.:
```xaml
<dialogHostAvalonia:DialogHost>
...
    <dialogHostAvalonia:DialogHost.PopupPositioner>
        <positioners:AlignmentDialogPopupPositioner HorizontalAlignment="Stretch" VerticalAlignment="Stretch" Margin="16"/>
    </dialogHostAvalonia:DialogHost.PopupPositioner>
</dialogHostAvalonia:DialogHost>
```
Would fit you expectations.

**Reply by bearyung** [reply]
Works like a charm! Thank you for your prompt response. :)

**Reply by bearyung** [reply]
> Hello, @bearyung
> 
> Seems like this is can be dome via positioners. This is a custom classes which can arrange dialoghost popup in some ways. E.g.:
> 
> ```
> <dialogHostAvalonia:DialogHost>
> ...
>     <dialogHostAvalonia:DialogHost.PopupPositioner>
>         <positioners:AlignmentDialogPopupPositioner HorizontalAlignment="Stretch" VerticalAlignment="Stretch" Margin="16"/>
>     </dialogHostAvalonia:DialogHost.PopupPositioner>
> </dialogHostAvalonia:DialogHost>
> ```
> 
> Would fit you expectations.

@SKProCH  I've encountered another issue after setting the AlignmentDialogPopupPositioner.

After I've setup that with Margin (e.g. Margin = 36), when the dialog content included an Avalonia Datagrid, all the right side of controls in the same view will not be able to show.

As illustrated in the following screenshot:
<img width="1166" alt="image" src="https://github.com/AvaloniaUtils/DialogHost.Avalonia/assets/13907070/fb741898-e52f-4243-826e-8af8c3f47c5a">

I've upload the repo to reproduce the issue:
https://github.com/bearyung/DialogHostDemo

Thank you very much for your help.

**Reply by SKProCH** [reply]
This is caused by your view can't be smaller that the width you specified in the columns. This is not a dialog specific bug. So, if you use your view inside a window directly it has the same behavior (also, it shows the scrollbar in the bottom).

**Reply by SKProCH** [reply]
Got your problem. Fixed with https://github.com/AvaloniaUtils/DialogHost.Avalonia/pull/54
Also, i've currently repairing pipelines, so release will be shortly (i hope).

**Reply by bearyung** [reply]
> Got your problem. Fixed with #54 Also, i've currently repairing pipelines, so release will be shortly (i hope).

Thanks for your prompt response! :) Cheers
